frontend/test/metabase/scenarios/setup/setup.cy.spec.js

```diff
@@ -1,143 +1,162 @@
 import path from "path";
 import { restore } from "__support__/cypress";
 
+// we're testing for one known (en) and one unknown (xx) locale
+const locales = ["en", "xx"];
```

Comment:
It would probably be better to do something that also tested a locale that isn't English, just to make sure that we're loading that language too. Example Portuguese and a nonexistent locale. That would make the test slightly bigger/more complex, but probably worth it. But let's see what Paul thinks, when he has a look.
```suggestion
const locales = ["pt", "xx"];
```

Once #13039 is merged to `master`, then #13048 will be included. All fixes related to 0.36 goes into `release-0.36.x` and after a release, then we merge back to `master` (upcoming 0.37)

Reply:
@nemanjaglumac thanks for doing this!

I agree including an available locale that's not `en` would be nice, but it probably requires updating the test to find localized strings. e.g. `cy.findByText("Welcome to Metabase");` would need to become `cy.findByText("[Welcome in Portuguese]")` or something. We could have a lookup for each string in the test:

```js
const testStrings = {
  en: {welcome: "Welcome to Metabase"},
  pt: {welcome: "[Welcome in Portuguese]"},
}
```

That would require updating the test file for changed translations. Hopefully these strings are pretty static, but if it becomes an issue we could create a fake locale with translations that we know won't change.

Another possible weirdness is updating the test to accommodate changes to the setup flow. If we added another step in the flow, we wouldn't immediately have the translations for that step. We'd need to use the English fallback until the files were updated.

All that said, it'd be nice to also test another available locale in addition to `en` and `xx`, but this is an improvement and addresses the issue from Friday. Merging this as-is once #13048 is in master or adding coverage for another language both sound good to me.

***

Another thought:
_Maybe_ we could use the ttag helper within Cypress? e.g. `` t`Welcome to Metabase` `` but with syntax that let's us call `t` and pass the local directly. That also feels a bit flawed since if `t` were broken and fell back to English, the tests would still pass.

Reply:
@paulrosenzweig You're welcome.
I was actually testing for more than `en` locally. Weird thing is - all tests for known locales passed although the interface language was still displayed in English. But, try to test for `da`, for example and it fails.

![image](https://user-images.githubusercontent.com/31325167/89217384-61dadf00-d5cc-11ea-9770-faf5cb6c91e2.png)

I was also testing in console and `navigator.locale` is always set to the actual language before the test begins (as it should). So, why is the setup page rendered in English?

In parallel, I opened another Metabase instance via Docker in another browser where I manually changed the browser language. It picks up the language there and the setup page is displayed localized.

Reply:
https://stackoverflow.com/questions/56791796/how-to-set-the-browsers-language-in-cypress-io-electron-chrome

> we tried overwriting navigator.locale with a shim during onBeforeLoad that returns our desired locale, but this doesn't change the locale using in GET requests.

> Note that this implements only one effect of changing the browser language - for example, this doesn't change the behavior of Date.toLocaleString, and the Accept-Language header will not be sent.

Reply:
@nemanjaglumac You've found a bug. It is supposed to show the interface according to the initial browser language (if that locale is available in Metabase), but it seems like only mouse-clicks on "Step 1" list actually changes the language.

Not sure why `xx` works - must be something in Chrome/Cypress that does something, since it's an illegal language code. Probably resets it to `en`.

Reply:
@flamber - `xx` "passes" because I've made a temporary exception for it (something like `it.skip()`) until Paul's fix is merged. Otherwise, it would've failed also.

Are you sure this is a bug and not just a Cypress quirk? It seems that there isn't a (simple) workaround in Cypress to do this. It definitely sets the `navigator.language` to the one we told it to, but the backend never receives it.

One thing this test definitely does - it reproduces issue #13045 . In other words, we can spot the bug that prevents the setup with an unknown locale, but I don't see the way we can test for localized versions of known locales. I hope it makes sense.

Reply:
Aaaaaand, one more update. I've deleted the local database in my dev environment to reset everything. Tried to open `localhost:3000` in Firefox with `fr` set as a language. It behaves exactly as you described:
https://share.getcloudapp.com/RBuOww02

Reply:
@nemanjaglumac Hmm, I just retested and now it sets it. Think it's a frontend state thing. I changed language after viewing the setup initially, and browser refresh won't do anything in that case. Could that be why you're having problems too?

Reply:
@flamber tbh, I don't know anymore. It could be. I have changed the browser language to from `fr` to `de` now, restarted Firefox and opened Docker version - it is displayed in English. o.O
- console doesn't show any errors
- on the next step the page is displayed in English as well, but the German is preselected as the preferred language

Reply:
@nemanjaglumac maybe your translation resources aren't built? I forget exactly when that's needed, but I could take a look later. If they're not there, the fallback would be to put everything in English. I can take a look later.

Reply:
@paulrosenzweig it definitey did happen with the dev environment (caught the missing resource error a few times).

But that is exactly why I did aditional, control test with the Docker. Unfortunately, in this last run, Docker Metabase failed to show setup page in German.